paymentsheet/src/main/java/com/stripe/android/paymentsheet/ui/PrimaryButtonNew.kt

```diff
@@ -0,0 +1,381 @@
+import androidx.compose.animation.core.tween
+import androidx.compose.animation.fadeIn
+import androidx.compose.animation.fadeOut
+import androidx.compose.animation.togetherWith
+import androidx.compose.foundation.BorderStroke
+import androidx.compose.foundation.isSystemInDarkTheme
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.BoxScope
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.IntrinsicSize
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.defaultMinSize
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material.ButtonDefaults
+import androidx.compose.material.ContentAlpha
+import androidx.compose.material.Icon
+import androidx.compose.material.LocalContentAlpha
+import androidx.compose.material.RadioButton
+import androidx.compose.material.Text
+import androidx.compose.material.TextButton
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.CompositionLocalProvider
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableFloatStateOf
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.BiasAlignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.platform.LocalInspectionMode
+import androidx.compose.ui.res.dimensionResource
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.TextStyle
+import androidx.compose.ui.text.font.Font
+import androidx.compose.ui.text.font.FontFamily
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import com.stripe.android.common.ui.LoadingIndicator
+import com.stripe.android.core.strings.ResolvableString
+import com.stripe.android.paymentsheet.R
+import com.stripe.android.uicore.StripeTheme
+import com.stripe.android.uicore.getBackgroundColor
+import com.stripe.android.uicore.getBorderStrokeColor
+import com.stripe.android.uicore.getOnBackgroundColor
+import kotlinx.coroutines.delay
+
+private const val FADE_ANIMATION_DURATION = 100
+private const val FADE_OUT_ANIMATION_DELAY = 90
+
+private const val PRE_SUCCESS_ANIMATION_DELAY = 250L
+private const val POST_SUCCESS_ANIMATION_DELAY = 1500L
+
+private const val TEXT_SUBDUED_VISIBILITY = 0.5f
+
+private val processCompleteBackground = Color(0xFF24B47E)
```

Comment:
Can we make this a color resource? Also, do we need different variants of this success color for light and dark mode?